### PR TITLE
Reduce planning memory usage with lots of chunks

### PR DESCRIPTION
**Pull request by akuzm**
Selectivity functions, like scalararrayinsel, allocate _a lot_ in the top planner context. Use a temporary memory context to call them.

Fixes https://github.com/timescale/timescaledb/issues/4944

### CODE REVIEW

File: src/planner/planner.c [hunk 1]
```diff
@@ -553,11 +570,21 @@ timescaledb_planner(Query *parse, int cursor_opts, ParamListInfo bound_params)
...
 													 /* private_data = */ NULL);
 			}
+
+			/*
+			 * Create a temporary memory context that can optionally be used by
+			 * the planner at this level of timescaledb_planner(). The inner
+			 * recursive invocations of timescaledb_planner() will also create
+			 * their own temporary contexts.
+			 */
+			MemoryContext mcxt = AllocSetContextCreate(CurrentMemoryContext,
```
Comment (akuzm) [inline src/planner/planner.c:584]: I'm kinda reluctant to add another weird cache based on static variables, and an entire stack this time... On the other hand, the general Postgres planning approach of "leaking memory like there's no tomorrow" ((c) the mailing lists) becomes problematic when you have a lot of partitions.

Comment (svenklemm) [reply src/planner/planner.c:584]: I 'm not a fan of adding nested memory contexts like this.

Comment (akuzm) [reply src/planner/planner.c:584]: Yeah, I don't like it either, feels fragile. We can just drop this as wontfix. Or maybe you have better ideas?

Comment (akuzm) [reply src/planner/planner.c:584]: The planning memory usage proportional to the number of chunks times the number of array elements also doesn't feel right, but not sure how to fix it in extension.

**Comment by akuzm** [comment]
Closing for now because this is an upstream problem, and the fix is potentially more problematic.

**Pull request closed without merging**